### PR TITLE
broadcast in flair-visualization

### PR DESCRIPTION
**Pull request by karthik-reddivari-dx26**

### CODE REVIEW

File: js/charts/stackedverticalbar.js [hunk 1]
```diff
@@ -1,16 +1,14 @@
...
 try {
-    var d3Lasso = require('../../d3-libs/d3-lasso.min.js');
-} catch (ex) { }
-
+    var d3Lasso = require("../../d3-libs/d3-lasso.min.js");
+} catch (ex) {}
 
 function stackedverticalbar() {
-
-    var _NAME = 'stackedverticalbar';
```
Comment (dev-fe-dx26) [inline js/charts/stackedverticalbar.js]: place it in constant file

File: js/charts/scatter.js [hunk 2]
```diff
@@ -1,16 +1,14 @@
...
 try {
-    var d3Lasso = require('../../d3-libs/d3-lasso.min.js');
-} catch (ex) { }
-
+    var d3Lasso = require("../../d3-libs/d3-lasso.min.js");
+} catch (ex) {}
 
 function scatter() {
-
-    var _NAME = 'scatterChart';
```
Comment (dev-fe-dx26) [inline js/charts/scatter.js:13]: place it in constant file

File: js/charts/bullet.js [hunk 3]
```diff
@@ -668,69 +753,97 @@ function bullet() {
...
+            )
+            .on(
+                "mousemove",
+                _handleMouseMoveFn.call(chart, tooltip, _local_svg)
+            )
+            .on("mouseout", _handleMouseOutFn.call(chart, tooltip, _local_svg))
+            .on("click", function (d) {
                 if (isLiveEnabled) {
-                    broadcast.$broadcast('FlairBi:livemode-dialog');
+                    broadcast.$broadcast("FlairBi:livemode-dialog");
```
Comment (dev-fe-dx26) [inline js/charts/bullet.js:802]: rename FlairBi to dx26-io

File: js/charts/bullet.js [hunk 4]
```diff
@@ -747,74 +860,109 @@ function bullet() {
...
-                } else if (orientation == 'Vertical') {
-                    return 'middle';
+        plot.selectAll(".title").remove();
+        var title = _bullet
+            .append("g")
+            .style("text-anchor", function (d) {
+                if (orientation == "Horizontal") {
+                    return "end";
+                } else if (orientation == "Vertical") {
+                    return "middle";
```
Comment (dev-fe-dx26) [inline js/charts/bullet.js]: Horizontal and Vertical should be in constant file

File: js/charts/bullet.js [hunk 5]
```diff
@@ -747,74 +860,109 @@ function bullet() {
...
+            .attr("class", "title")
+            .style("font-style", fontStyle)
+            .style("font-weight", fontWeight)
+            .style("font-size", fontSize + "px")
+            .attr("transform", function (d) {
+                if (orientation == "Horizontal") {
+                    return "rotate(0)";
+                } else if (orientation == "Vertical") {
+                    return "rotate(90)";
                 }
```
Comment (dev-fe-dx26) [inline js/charts/bullet.js]: place it in constant file

File: js/charts/chorddiagram.js [hunk 6]
```diff
@@ -138,150 +163,147 @@ function chorddiagram() {
...
+            )
+            .on(
+                "mousemove",
+                _handleMouseMoveFn.call(chart, tooltip, _local_svg)
+            )
+            .on("mouseout", _handleMouseOutFn.call(chart, tooltip, _local_svg))
+            .on("click", function (d, i) {
                 if (isLiveEnabled) {
-                    broadcast.$broadcast('FlairBi:livemode-dialog');
+                    broadcast.$broadcast("FlairBi:livemode-dialog");
```
Comment (dev-fe-dx26) [inline js/charts/chorddiagram.js:285]: rename flair-bi

File: js/charts/chorddiagram.js [hunk 7]
```diff
@@ -590,121 +643,143 @@ function chorddiagram() {
...
-                }
-                else if (_colorPattern == 'gradient_color') {
+                } else if (_colorPattern == "gradient_color") {
                     var value = gdata.filter(function (val) {
                         if (val["key"] == d) {
                             return val;
                         }
-                    })
+                    });
                     return gradientColor(value[0].value);
```
Comment (dev-fe-dx26) [inline js/charts/chorddiagram.js:704]: if it's an array rename value to values

File: js/charts/clusteredhorizontalbar.js [hunk 8]
```diff
@@ -127,60 +124,94 @@ function clusteredhorizontalbar() {
...
-            "<th>" + datum["measure"] + ": </th>" +
-            "<td>" + UTIL.getFormattedValue(datum[datum["measure"]], UTIL.getValueNumberFormat(_measure.indexOf(datum["measure"]), _numberFormat, datum[datum["measure"]])) + " </td>" +
+            "<th>" +
+            datum["measure"] +
+            ": </th>" +
+            "<td>" +
+            UTIL.getFormattedValue(
+                datum[datum["measure"]],
+                UTIL.getValueNumberFormat(
+                    _measure.indexOf(datum["measure"]),
```
Comment (dev-fe-dx26) [inline js/charts/clusteredhorizontalbar.js]: _measure if it is an array rename _measures

File: js/charts/clusteredhorizontalbar.js [hunk 9]
```diff
@@ -221,185 +255,231 @@ function clusteredhorizontalbar() {
...
                 var _filterDimension = {};
-                if (broadcast.filterSelection.id) {
-                    _filterDimension = broadcast.filterSelection.filter;
-                } else {
-                    broadcast.filterSelection.id = parentContainer.attr('id');
-                }
-                var dimension = _dimension[0];
 
-                _filterDimension[dimension] = filterData.map(function (d) {
+                _filterDimension[_dimension[0]] = filterData.map(function (d) {
```
Comment (dev-fe-dx26) [inline js/charts/clusteredhorizontalbar.js]: rename _dimension to _dimensions

File: js/charts/clusteredverticalbar.js [hunk 10]
```diff
@@ -110,84 +121,120 @@ function clusteredverticalbar() {
...
+            config.measure,
+            config.displayNameForMeasure
+        );
+    };
 
     var setDefaultColorForChart = function () {
         for (let index = 0; index < _measure.length; index++) {
-            if (_displayColor[index] == null || _displayColor[index] == undefined) {
+            if (
+                _displayColor[index] == null ||
```
Comment (dev-fe-dx26) [inline js/charts/clusteredverticalbar.js]: only if(_displayColor[index] ) is enough?

**Review by dev-fe-dx26 — CHANGES_REQUESTED** [review]
1)put hardcoded values in constant file
2)if it's array then use name accordingly for example if its array of measure then it should measures instead of measure

File: js/charts/boxplot.js [hunk 11]
```diff
@@ -4,7 +4,7 @@ var UTIL = require("../extras/util.js")();
 var $ = require("jquery");
 try {
     var d3Lasso = require("../../d3-libs/d3-lasso.min.js");
-} catch (ex) {}
+} catch (ex) { }
 function boxplot() {
     var _NAME = "boxplot";
```
Comment (dev-fe-dx26) [inline js/charts/boxplot.js]: put this in constant file

File: js/charts/bullet.js [hunk 12]
```diff
@@ -5,7 +5,7 @@ var UTIL = require("../extras/util.js")();
 var $ = require("jquery");
 try {
     var d3Lasso = require("../../d3-libs/d3-lasso.min.js");
-} catch (ex) {}
+} catch (ex) { }
 function bullet() {
     var _NAME = "bullet";
```
Comment (dev-fe-dx26) [inline js/charts/bullet.js]: this also

File: js/charts/chorddiagram.js [hunk 13]
```diff
@@ -6,7 +6,7 @@ var viz = require("../../d3-libs/viz.js");
 var $ = require("jquery");
 try {
     var d3Lasso = require("../../d3-libs/d3-lasso.min.js");
-} catch (ex) {}
+} catch (ex) { }
 
 function chorddiagram() {
     var _NAME = "chorddiagram";
```
Comment (dev-fe-dx26) [inline js/charts/chorddiagram.js]: this also

File: js/charts/combo.js [hunk 14]
```diff
@@ -5,7 +5,7 @@ var LEGEND = require("../extras/legend_barcharts.js")();
 var $ = require("jquery");
 try {
     var d3Lasso = require("../../d3-libs/d3-lasso.min.js");
-} catch (ex) {}
+} catch (ex) { }
 
 function combo() {
     var _NAME = "combo";
```
Comment (dev-fe-dx26) [inline js/charts/combo.js]: this also

File: js/charts/heatmap.js [hunk 15]
```diff
@@ -5,7 +5,7 @@ var $ = require("jquery");
 
 try {
     var d3Lasso = require("../../d3-libs/d3-lasso.min.js");
-} catch (ex) {}
+} catch (ex) { }
 function heatmap() {
     var _NAME = "heatmap";
```
Comment (dev-fe-dx26) [inline js/charts/heatmap.js]: this also

File: js/charts/map.js [hunk 16]
```diff
@@ -7,7 +7,7 @@ var $ = require("jquery");
 
 try {
     var d3Lasso = require("../../d3-libs/d3-lasso.min.js");
-} catch (ex) {}
+} catch (ex) { }
 function map() {
     var _NAME = "map";
```
Comment (dev-fe-dx26) [inline js/charts/map.js]: this one

**Review by dev-fe-dx26 — CHANGES_REQUESTED** [review]

File: js/charts/stackedverticalbar.js (hunk 1 above)
Comment (karthik-reddivari-dx26) [reply js/charts/stackedverticalbar.js]: done

File: js/charts/scatter.js (hunk 2 above)
Comment (karthik-reddivari-dx26) [reply js/charts/scatter.js:13]: done

File: js/charts/bullet.js (hunk 3 above)
Comment (karthik-reddivari-dx26) [reply js/charts/bullet.js:802]: its not needed now

File: js/charts/chorddiagram.js (hunk 6 above)
Comment (karthik-reddivari-dx26) [reply js/charts/chorddiagram.js:285]: its not needed now

File: js/charts/boxplot.js (hunk 11 above)
Comment (karthik-reddivari-dx26) [reply js/charts/boxplot.js]: done

File: js/charts/bullet.js (hunk 4 above)
Comment (karthik-reddivari-dx26) [reply js/charts/bullet.js]: done

File: js/charts/bullet.js (hunk 5 above)
Comment (karthik-reddivari-dx26) [reply js/charts/bullet.js]: done

File: js/charts/chorddiagram.js (hunk 7 above)
Comment (karthik-reddivari-dx26) [reply js/charts/chorddiagram.js:704]: it is one value filtering from array

File: js/charts/clusteredhorizontalbar.js (hunk 9 above)
Comment (karthik-reddivari-dx26) [reply js/charts/clusteredhorizontalbar.js]: done

File: js/charts/clusteredverticalbar.js (hunk 10 above)
Comment (karthik-reddivari-dx26) [reply js/charts/clusteredverticalbar.js]: done

File: js/charts/bullet.js (hunk 3 above)
Comment (dev-fe-dx26) [reply js/charts/bullet.js:802]: why? this is dx26 project right

File: js/charts/chorddiagram.js (hunk 6 above)
Comment (dev-fe-dx26) [reply js/charts/chorddiagram.js:285]: check the above comment

File: js/charts/chorddiagram.js (hunk 7 above)
Comment (dev-fe-dx26) [reply js/charts/chorddiagram.js:704]: ok

**Pull request merged**